packages/cadl-ranch-specs/http/overload/main.cadl

```diff
@@ -0,0 +1,74 @@
+import "@cadl-lang/rest";
+import "@azure-tools/cadl-ranch-expect";
+
+using Cadl.Http;
+
+@doc("Overload")
+@supportedBy("dpg")
+@scenarioService("/overload")
+namespace Overload;
+
+@scenario
+@scenarioDoc("""
+send "hello world" with `conten-type` "text/plain";
+send binary data "hello world" with `conten-type` "application/octet-stream"
+""")
+@route("/bytesOrString")
+@post
+op uploadBytesOrString(
```

Comment:
Should we have some way to mark this as "don't use" for the clients?  This isn't typically what `overload` means.  In this case we are trying to use `overload` as `overwrite with 1 or more items`.  I prefer we change the decorator or have a way to mark the original as "don't use".

@johanste @lmazuel @srnagar 

There was also discussion to use a different decorator like allow duplicate routes but I can't seem to find any documentation on it.  @srnagar do you remember the name of that decorator or have a link to the documentation?

Reply:
Hi @m-nash The testcase is to test the [scenario1](https://gist.github.com/msyyc/87a79b917deec25639f9f10d20589e55#scenario-1) and [scenario2](https://gist.github.com/msyyc/87a79b917deec25639f9f10d20589e55#scenario-2). The testcase is limited by cadl-ranch so I think the scenario doc is much clear. 
In the scenario1 and 2, author could define very clear map for body type and content type which gives language emitter enough info to generate user-friendly SDK code like [here](https://github.com/Azure/autorest.python/blob/6b2c21e5b96ca9284fa8b9edc7f407c18774390d/packages/cadl-python/test/mock_api_tests/test_overload.py#L16-L17). With overload, users don't need to input `content-type` since SDK can judge it with input type which the abosolutely `overload` means.

Reply:
If only python decides to support the `overload` thing, I would suggest put this spec and corresponding mockapi into python codegen repo. You just need to pass the mockapi path in your codegen repo to cadl-ranch server, and it could serve as a service in your own repo. Example: 
Mockapi: https://github.com/Azure/autorest.csharp/blob/2ff04cd6760e4371dc031d2022c3b43440bdb101/test/CadlRanchMockApis/src/Lro-Basic-Cadl.ts
Spec:
https://github.com/Azure/autorest.csharp/blob/2ff04cd6760e4371dc031d2022c3b43440bdb101/test/TestProjects/Lro-Basic-Cadl/Lro-Basic-Cadl.cadl
Test:
https://github.com/Azure/autorest.csharp/blob/2ff04cd6760e4371dc031d2022c3b43440bdb101/test/CadlRanchProjects.Tests/LroBasic-Cadl.cs

Reply:
@m-nash are you proposing to decorate the "general/overloaded" function with `do not use` to indicate emitters that they shouldn't generate that function but just the overloads? Or just mark them internally in the "code model" for emitters to decide what to do? If it is the latter I agree, I'm not sure I like the decorator idea.

FWIW this is the same pattern that [TypeScript uses for overloading](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads), where the overloaded function is not exposed, but we do need to generate it.